lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -102,13 +193,32 @@ class DomNavigator {}
 
 extension DomNavigatorExtension on DomNavigator {
   external DomClipboard? get clipboard;
-  external double? get maxTouchPoints;
-  external String get vendor;
-  external String get language;
-  external String? get platform;
-  external String get userAgent;
-  List<String>? get languages =>
-      js_util.getProperty<List<Object?>?>(this, 'languages')?.cast<String>();
+
+  @JS('maxTouchPoints')
+  external JSNumber? get _maxTouchPoints;
+  double? get maxTouchPoints => _maxTouchPoints?.toDart;
```

Comment:
With this new approach, it looks like we can start using `int` again? (maybe with a `.toInt()` at the end there)

Reply:
Yes, we can start using `int` again now that we are handling the conversions at the boundary.

The rule going forward is 'if the function is external, use JS types`, and beyond that we can do whatever makes the most sense. Once I migrate the framework and `package:test` then this rule will be enforced at compile time in Dart2Wasm, but for the next few weeks we just have to be careful. It won't break anything to violate this rule, it will just make flipping the 'strict mode' flag for Dart2Wasm a bit harder.

We can do a cleanup pass after this to make the engine facing side of the DOM shim more ergonomic.